### ISSUE TITLE
Inquiry about Preprocessing Discrepancies in ONNX Runtime C++ YOLO-CLS Implementation

### ISSUE THREAD

**Issue by SongPool** [ultralytics/ultralytics#22128]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I’ve noticed that the image preprocessing sequence in the ONNX Runtime C++ implementation for YOLO-CLS appears to differ from the PyTorch version. Specifically:

Processing Order:
In the PyTorch version, the sequence is resize → crop.
In the ONNX Runtime C++ version, the sequence is crop → resize.
Interpolation Method:
The PyTorch version uses torchvision‘s InterpolationMode.BILINEAR as the default for resizing.
The ONNX Runtime C++ version uses OpenCV’s cv2.INTER_LINEAR.
These differences in interpolation methods may impact the final results. Could you clarify if this is intentional or if there are plans to align the preprocessing logic with the PyTorch implementation?

### Additional

<img width="1311" height="593" alt="Image" src="https://github.com/user-attachments/assets/bf613f30-f20f-4adf-905a-a63dc3c0bc7d" />

<img width="1872" height="932" alt="Image" src="https://github.com/user-attachments/assets/4a3d62cc-003a-407a-a095-61a155b09a33" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SongPool, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get support quickly—an Ultralytics engineer will also assist you here soon.

From your description, it sounds like a potential preprocessing mismatch between PyTorch and ONNX Runtime C++ for YOLO-CLS. To help us confirm and prioritize, please share a concise, side-by-side comparison of the exact preprocessing steps you’re using in both environments and any export details.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. For this case, an effective MRE would include:
- A single test image used in both pipelines
- The exact PyTorch preprocessing code (resize/crop order, interpolation, normalization)
- The exact ONNX Runtime C++ preprocessing code (resize/crop order, interpolation, normalization)
- Model export command/arguments (e.g., opset, image size, preprocessing flags)
- Package versions and environment details: ultralytics, PyTorch/torchvision, OpenCV, onnxruntime, OS, and hardware
- The observed outputs from both pipelines for the same image

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The code in the `ultralytics/examples` directory is community contributed and not always correct
